libr/anal/p/anal_x86_cs.c

```diff
@@ -1462,8 +1475,8 @@ static void anop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len,
 			// Set OF, SF, ZF, AF, PF, and CF flags.
 			// We use $b rather than $c here as the carry flag really
 			// represents a "borrow"
-			esilprintf (op, "%s,%s,%d,$o,of,:=,%d,$s,sf,:=,$z,zf,:=,$p,pf,:=,%d,$b,cf,:=",
-				src, dst, bitsize - 1, bitsize - 1, bitsize);
+			esilprintf (op, "%s,%s,%s,%u,-,!,%d,$o,^,of,:=,%d,$s,sf,:=,$z,zf,:=,$p,pf,:=,%d,$b,cf,:=,3,$b,af,:=",
+				src, dst, src, (1U << (bitsize - 1)), bitsize - 1, bitsize - 1, bitsize);
```

Comment:
bitsize can overflow this shift (undefined behaviour)

Reply:
is there a preference between using (ut64)1 or 1ULL? Other places use 1LL.

Reply:
Better use 1ULL imho